internal/ingest/ingest.go

```diff
@@ -203,6 +205,8 @@ func NewIngester(cfg config.Ingest, h host.Host, idxr indexer.Interface, reg *re
 	if cfg.IngestWorkerCount == 0 {
 		return nil, errors.New("ingester worker count must be > 0")
 	}
+
+	ing.workersCtx, ing.cancelWorkers = context.WithCancel(context.Background())
```

Comment:
`ing.cancelWorkers` never called?

Reply:
Oops, I meant to move it to the top of Close(), there now.